tree/tree/src/TTreeCloner.cxx

```diff
@@ -691,13 +729,28 @@ void TTreeCloner::WriteBaskets()
       TBranch *from = (TBranch*)fFromBranches.UncheckedAt( fBasketBranchNum[ fBasketIndex[j] ] );
       TBranch *to   = (TBranch*)fToBranches.UncheckedAt( fBasketBranchNum[ fBasketIndex[j] ] );
 
-      TFile *tofile = to->GetFile(0);
+      TFile *tofile = fToFile;
       TFile *fromfile = from->GetFile(0);
 
       Int_t index = fBasketNum[ fBasketIndex[j] ];
 
       Long64_t pos = from->GetBasketSeek(index);
-      if (pos!=0) {
+      if (IsInPlace()) {
+         if (pos != 0) {
```

Comment:
`pos == 0` cannot happen for `IsInPlace()`? If so, please add an `Error` message to diagnose this case should it ever happen. If `pos == 0` just doesn't need to do anything (because source == target data?) then please add a small comment to help a lost soul having to review this code!